internal/storage/image.go

```diff
@@ -309,6 +311,30 @@ func (svc *imageService) buildImageResult(image *storage.Image, cacheItem imageC
 			break
 		}
 	}
+
+	// Try to retrieve the mountpoint
+	mountPoint := ""
+	if layer, err := svc.store.Layer(image.TopLayer); err == nil {
+		mountPoint = layer.MountPoint
+	} else {
+		logrus.Errorf("Unable to get image (%s) top layer (%s): %v", image.ID, image.TopLayer, err)
+	}
+	// Check if the mount actually exists
+	if mountPoint != "" {
+		infos, err := mountinfo.GetMounts(mountinfo.SingleEntryFilter(mountPoint))
+		if err != nil {
+			logrus.Warnf("Unable to get mount info for path %s: %v", mountPoint, err)
+			mountPoint = ""
+		}
+		if len(infos) == 0 {
+			logrus.Warnf("Mount path %s for image (%s) is not mounted any more, unmounting it in c/storage", mountPoint, image.ID)
```

Comment:
What about something like this, an example:

> Unable to find mount path %s for image %s, assuming image is no longer mounted

or

> Mount path %s is no longer present, assuming image %s is no longer mounted

The idea is not to expose the implementation details too much to the end user. I assume that "unmounting it in c/storage" means cleaning up some internal state? Otherwise, if c/storage tracks OS-level mounts, then it should also notice the mount point is gone, I assume.

Reply:
Sure, adapted as suggested :+1: